### DISCUSSION TITLE
Rendering markdown inside specific sections

### DISCUSSION THREAD

**Discussion by merfed** [thephpleague/commonmark#1022] [Q&A] [answered]
I have a extension I created that creates a banner, it has 3 renderers: root banner element, banner image, and banner caption. The syntax is `~[caption](image_url)`. Everything works as I expect, except for one thing, how can I get markdown to parse in the caption. Such as `~[_test_](image_url)`. It's simply outputting "_test_" as the caption instead of converted markdown.

I've looked through the docs, but maybe I missed something.

**Comment by samwilson** [comment]
> It's simply outputting "_test_" as the caption

I assume you meant this to say `outputting "_test_" as the caption` rather than showing the italics here? :)

But it sounds like you need to recursively parse the caption contents, probably with the InlinesOnlyExtension extension enabled.

**Comment by merfed** [comment]
This is what I have:

```
<?php

namespace App\Extensions\Banner;

use League\CommonMark\Extension\CommonMark\Node\Inline\Image;
use League\CommonMark\Node\Inline\Text;
use League\CommonMark\Parser\Inline\InlineParserInterface;
use League\CommonMark\Parser\Inline\InlineParserMatch;
use League\CommonMark\Parser\InlineParserContext;

final class BannerInlineParser implements InlineParserInterface
{
    public function getMatchDefinition(): InlineParserMatch
    {
        return InlineParserMatch::regex('^~\[([^~\]]*)\]\(([^~\]]*)\)');
    }

    public function parse(InlineParserContext $inlineContext): bool
    {
        $cursor = $inlineContext->getCursor();

        if (! $cursor->match($this->getMatchDefinition()->getRegex())) {
            return false;
        }

        [$caption, $url] = $inlineContext->getSubMatches();

        $cursor->advanceBy($inlineContext->getFullMatchLength());

        $bannerNode = new BannerNode();

        $image = new Image($url);
        $bannerNode->appendChild($image);
        
        if ($caption !== '') {
            $captionNode = new BannerCaption();
            $captionNode->appendChild(new Text($caption));

            $bannerNode->appendChild($captionNode);
        }

        $inlineContext->getContainer()->appendChild($bannerNode);

        return true;
    }
}
```

Should text inside the Text node, not get parsed by the markdown parser? That seems to be the case.

**Comment by colinodell** [comment, marked as answer]
Because your parser follows similar syntax to links and images, I'd recommend looking at [how our CloseBracketParser handles this by processing the delimiters](https://github.com/thephpleague/commonmark/blob/6fbb36d44824ed4091adbcf4c7d4a3923cdb3405/src/Extension/CommonMark/Parser/Inline/CloseBracketParser.php#L104-L109) (syntax like `*` and `_` that wraps/delimits text). You should be able to do something similar.
